biodatasets/pubtator_central/pubtator_central.py

```diff
@@ -0,0 +1,322 @@
+including warranties of performance, merchantability or fitness for any particular purpose.
+"""
+
+_URLS = {_DATASETNAME: "https://ftp.ncbi.nlm.nih.gov/pub/lu/PubTatorCentral/bioconcepts2pubtatorcentral.offset.sample"}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION, Tasks.NAMED_ENTITY_DISAMBIGUATION]
+
+_SOURCE_VERSION = "2022.01.08"
+_BIGBIO_VERSION = "1.0.0"
+
+# Maps the entity types in PubTator to the name of the database they are grounded to
+_TYPE_TO_DB_NAME = {
+    "Gene": "ncbi_gene",
+    "Disease": "mesh",
+    "Species": "ncbi_taxon",
+    "Chemical": "mesh",
+    "CellLine": "cellosaurus",
+}
+
+_DB_NAME_TO_URL = {
+    "ncbi_gene": "https://www.ncbi.nlm.nih.gov/gene/",
+    "mesh": "https://www.nlm.nih.gov/mesh/meshhome.html",
+    "ncbi_taxon": "https://www.ncbi.nlm.nih.gov/taxonomy/",
+    "cellosaurus": "https://web.expasy.org/cellosaurus/",
+    "ncbi_dbsnp": "https://www.ncbi.nlm.nih.gov/snp/",
+    "tmvar": "https://www.ncbi.nlm.nih.gov/research/bionlp/Tools/tmvar/",
+}
+
+
+class PubtatorCentralDataset(datasets.GeneratorBasedBuilder):
+    """PubTator Central"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="pubtator_central_source",
+            version=SOURCE_VERSION,
+            description="PubTator Central source schema",
+            schema="source",
+            subset_id="pubtator_central",
+        ),
+        BigBioConfig(
+            name="pubtator_central_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="PubTator Central BigBio schema",
+            schema="bigbio_kb",
+            subset_id="pubtator_central",
+        # sample source 
+        BigBioConfig(
+            name="pubtator_central_sample_source",
+            version=SOURCE_VERSION,
+            description="PubTator Central sample source schema",
+            schema="source",
+            subset_id="pubtator_central_sample",
+        ),
+        # sample big bio
+        BigBioConfig(
+            name="pubtator_central_sample_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="PubTator Central sample BigBio schema",
+            schema="bigbio_kb",
+            subset_id="pubtator_central_sample",
+        ),
+        # full dataset source
+        BigBioConfig(
+            name="pubtator_central_source",
+            version=SOURCE_VERSION,
+            description="PubTator Central source schema",
+            schema="source",
+            subset_id="pubtator_central",
+        ),
+        # full dataset bigbio
+        BigBioConfig(
+            name="pubtator_central_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="PubTator Central BigBio schema",
+            schema="bigbio_kb",
+            subset_id="pubtator_central",
+        ),
```

Comment:
sorry for the confusion here, but i was trying to recommend 4 configs not 6

Reply:
No worries, these are the four right? ({source, bigbio} * {sample, full}):

```python
    BUILDER_CONFIGS = [
        # sample source
        BigBioConfig(
            name="pubtator_central_sample_source",
            version=SOURCE_VERSION,
            description="PubTator Central sample source schema",
            schema="source",
            subset_id="pubtator_central_sample",
        ),
        # sample big bio
        BigBioConfig(
            name="pubtator_central_sample_bigbio_kb",
            version=BIGBIO_VERSION,
            description="PubTator Central sample BigBio schema",
            schema="bigbio_kb",
            subset_id="pubtator_central_sample",
        ),
        # full dataset source
        BigBioConfig(
            name="pubtator_central_source",
            version=SOURCE_VERSION,
            description="PubTator Central source schema",
            schema="source",
            subset_id="pubtator_central",
        ),
        # full dataset bigbio
        BigBioConfig(
            name="pubtator_central_bigbio_kb",
            version=BIGBIO_VERSION,
            description="PubTator Central BigBio schema",
            schema="bigbio_kb",
            subset_id="pubtator_central",
        ),
    ]
```

Reply:
yep